app/models/group.rb

```diff
@@ -26,6 +26,8 @@ class Group < ActiveRecord::Base
   has_many :category_reviews, class_name: 'Category', foreign_key: :reviewable_by_group_id, dependent: :nullify
   has_many :reviewables, foreign_key: :reviewable_by_group_id, dependent: :nullify
 
+  belongs_to :flair_image, class_name: 'Upload'
```

Comment:
This association seems strange to me. I would think that a `Group` -> `has_one :flair_image` rather than `belongs_to` it

Reply:
Since the target table has primary key field (`upload.id`) and "group" table has child key we should use `belongs_to` here, I hope. We have a similar code in [`user.rb`](https://github.com/discourse/discourse/blob/6c1c8be79433f87bef9d768da7b8fa4ec9bb18d7/app/models/user.rb#L53) file.

Previously, I tried `has_one` and it returned an error.

Reply:
Yup you’re right. I got confused about where the foreign key should be on.